store/splitstore.go

```diff
@@ -0,0 +1,93 @@
+package store
+
+import (
+	"github.com/loomnetwork/go-loom/plugin"
+)
+
+type splitStore struct {
+	KVReader
+	VersionedKVStore
+	deleted map[string]bool
+	version int64
+}
+
+func NewSplitStore(full KVReader, empty VersionedKVStore, version int64) VersionedKVStore {
+	return &splitStore{
+		KVReader:         full,
+		VersionedKVStore: empty,
+		deleted:          make(map[string]bool),
+		version:          version,
+	}
+}
+
+func (ss *splitStore) Get(key []byte) []byte {
+	if ss.VersionedKVStore.Has(key) {
+		return ss.VersionedKVStore.Get(key)
+	}
+	if ss.deleted[string(key)] {
+		return nil
+	}
+	return ss.KVReader.Get(key)
+}
+
+func (ss *splitStore) Range(prefix []byte) plugin.RangeData {
+	readerRange := ss.KVReader.Range(prefix)
+	updateRange := ss.VersionedKVStore.Range(prefix)
```

Comment:
Seeing as `ss.VersionedKVStore` is really a `MemStore` and `MemStore.Range` iterates over a `map`... this is going to be non-deterministic.

Reply:
Added a sort to splitstore.Range. I could add the sort to MemStore.Range?